### DISCUSSION TITLE
Raspberry Pi (ARM) docker images now on available on Docker Hub!

### DISCUSSION THREAD

**Discussion by jha749** [DeFiCh/app#939] [Show and tell]
*this is a copy of my post on Reddit [here](https://www.reddit.com/r/defiblockchain/comments/nsuo2d/raspberry_pi_arm_docker_images_now_on_available/), I welcome feedback both here and on reddit!*

Hi all,

For everyone running or wanting to run a Defichain node on a Raspberry Pi, ARM 32-bit docker images can now be found directly in Docker Hub. New releases of Defichain are automatically built and pushed to DockerHub (with a max delay of 1 day). I'm running this myself on a Raspberry Pi4 (have been running a Pi node for over 2 months now). Having pre-built images available saves everyone wanting to run a Pi node a lot of hassle and time on compiling!

If you want to try it out, you can find the images here: [https://hub.docker.com/r/jha749/defichain/tags](https://hub.docker.com/r/jha749/defichain/tags)

To run it, you can use the official DefiChain instructions on running a docker node [here](https://github.com/DeFiCh/ain/blob/master/doc/setup-nodes-docker.md). Instead of pulling from defi/defichain you should pull from jha749/defichain, e.g. `docker pull jha749/defichain`.

I would welcome feedback!

Some notes:

* You can find the Dockerfile used [here](https://github.com/jha749/DefiChain-packager/blob/main/dockerfiles/arm32v7.dockerfile), that same repo contains the Github actions workflow that does the building.
* Ultimately, when I get some confirmation that it's generally running ok for more users, I plan to open a PR and see if the ARM dockerfile (and automatic build) can be integrated in the main project.
* I have had some memory errors in the past, for me it seems to work with having the following flags in defid command  `-maxmempool=64 -dbcache=128`. To make docker auto restart when the node somehow exits with an error, I use `--restart on-failure` in the `docker run` command (or `restart: on-failure` when using docker-compose).
* I want to add ARM64 builds as well, but then I first need someone with a 64-bit install on his/her Pi to test it. There is a test image available [here](https://hub.docker.com/r/jha749/test-defichain)
* I recommend adding a SSD or external HDD to your Pi to store the blockchain (using a WD elements external HDD myself with success). The SD cards in Pis have a name to get corrupted easily when you do a lot of read/write operations on them.
* If anyone is interested in trying it out but does not feel confident enough with bash / docker, let me know, I will try to create a step-by-step tutorial on setting it up.

**Comment by Martin8617** [comment]
You are one step in front of me.
Thanks for sharing your docker image.
On the other side I made the Raspberry Pi (armv7i) Defi-Wallet AppImage without docker , means exact the self app like on Windows, Mac and Linux.
Do you know a way to distribute it (yours or mines) on defich/app/releases?

**Comment by bet2019** [comment]
int64_t GetTimeMicros(): Assertion `now > 0' failed
